dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPIImpl.java

```diff
@@ -0,0 +1,163 @@
+package com.dotcms.experiments.business;
+
+import com.dotcms.business.CloseDBIfOpened;
+import com.dotcms.business.WrapInTransaction;
+import com.dotcms.contenttype.exception.NotFoundInDbException;
+import com.dotcms.experiments.model.AbstractExperiment.Status;
+import com.dotcms.experiments.model.Experiment;
+import com.dotcms.util.DotPreconditions;
+import com.dotcms.util.LicenseValiditySupplier;
+import com.dotmarketing.beans.PermissionableProxy;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.DotStateException;
+import com.dotmarketing.business.FactoryLocator;
+import com.dotmarketing.business.PermissionAPI;
+import com.dotmarketing.business.PermissionLevel;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.exception.InvalidLicenseException;
+import com.dotmarketing.portlets.contentlet.business.ContentletAPI;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.UUIDGenerator;
+import com.dotmarketing.util.UtilMethods;
+import com.liferay.portal.model.User;
+import java.time.Instant;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+public class ExperimentsAPIImpl implements ExperimentsAPI {
+
+    final ExperimentsFactory factory = FactoryLocator.getExperimentsFactory();
+    final PermissionAPI permissionAPI = APILocator.getPermissionAPI();
+    final ContentletAPI contentletAPI = APILocator.getContentletAPI();
+
+    private final LicenseValiditySupplier licenseValiditySupplierSupplier =
+            new LicenseValiditySupplier() {};
+
+    private final Supplier<String> invalidLicenseMessageSupplier =
+            ()->"Valid License is required";
+
+    @Override
+    @WrapInTransaction
+    public Experiment save(final Experiment experiment, final User user) throws
+            DotSecurityException, DotDataException {
+        DotPreconditions.isTrue(hasValidLicense(), InvalidLicenseException.class,
+                invalidLicenseMessageSupplier);
+
+        final Contentlet pageAsContent = contentletAPI
+                .findContentletByIdentifierAnyLanguage(experiment.pageId(), false);
+
+        DotPreconditions.isTrue(pageAsContent!=null
+                && UtilMethods.isSet(pageAsContent.getIdentifier()),
+                DotStateException.class, ()->"Invalid Page provided");
+
+        if(!permissionAPI.doesUserHavePermission(pageAsContent, PermissionLevel.EDIT.getType(), user)) {
```

Comment:
Because in this case I am loading the page before to validate that it is actually a page, and since I have the page it doesn't make sense to call validatePermissions, because that is intended to be called with only pageId and create a PermissionableProxy.